### PR TITLE
[ML] Upgrade restclient to v2022_10_01_preview

### PR DESCRIPTION
**Pull request by TonyJ1**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/automl_vertical.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
 
-from azure.ai.ml._restclient.v2022_06_01_preview.models import LogVerbosity
+from azure.ai.ml._restclient.v2022_10_01_preview.models import LogVerbosity
```
Comment (MaurisLucis) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/automl/automl_vertical.py:5]: I notice you're making changes to the _schema folder in this repo. Is _schema not just for CLI? I thought CLI changes were supposed to be made in the pre-migration repo.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_service.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 import logging
 from typing import Dict, Optional, Union
 
-from azure.ai.ml._restclient.v2022_06_01_preview.models import JobService as RestJobService20220601Preview
+from azure.ai.ml._restclient.v2022_10_01_preview.models import JobService as RestJobService20220601Preview
```
Comment (MaurisLucis) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/job_service.py:8]: import changed, name was not

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_util.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 from marshmallow.exceptions import ValidationError
 
 from azure.ai.ml._restclient.v2022_02_01_preview.models import JobInputType as JobInputType02
-from azure.ai.ml._restclient.v2022_06_01_preview.models import JobInputType as JobInputType06
+from azure.ai.ml._restclient.v2022_10_01_preview.models import JobInputType as JobInputType06
```
Comment (MaurisLucis) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_util.py:16]: looks like the suffix on the name may need to be updated as well

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_code_operations.py [hunk 4]
```diff
@@ -18,7 +18,7 @@
 from azure.ai.ml._restclient.v2021_10_01_dataplanepreview import (
     AzureMachineLearningWorkspaces as ServiceClient102021Dataplane,
 )
-from azure.ai.ml._restclient.v2022_06_01_preview import AzureMachineLearningWorkspaces as ServiceClient062022
+from azure.ai.ml._restclient.v2022_10_01_preview import AzureMachineLearningWorkspaces as ServiceClient062022
```
Comment (MaurisLucis) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_code_operations.py:21]: update name

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py [hunk 5]
```diff
@@ -131,15 +131,15 @@ def __init__(
...
+        service_client_10_2022_preview: ServiceClient102022Preview,
         all_operations: OperationsContainer,
         credential: TokenCredential,
         **kwargs: Any,
     ):
         super(JobOperations, self).__init__(operation_scope, operation_config)
         ops_logger.update_info(kwargs)
-        self._operation_2022_06_preview = service_client_06_2022_preview.jobs
-        self._service_client = service_client_06_2022_preview
+        self._operation_2022_06_preview = service_client_10_2022_preview.jobs
```
Comment (MaurisLucis) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py:141]: update attribute name

Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_job_operations.py]: Done

File: sdk/ml/azure-ai-ml/tests/automl_job/unittests/test_automl_image_schema.py [hunk 6]
```diff
@@ -121,18 +118,6 @@ def expected_image_target_column_name() -> str:
     return "label"
 
 
-@pytest.fixture
-def expected_image_sweep_settings() -> RestImageSweepSettings:
-    return RestImageSweepSettings(
-        limits=RestImageSweepLimitSettings(),
```
Comment (sharma-riti) [inline sdk/ml/azure-ai-ml/tests/automl_job/unittests/test_automl_image_schema.py:127]: I think we only need to remove the restimagesweeplimitsettings here?

Comment (TonyJ1) [reply sdk/ml/azure-ai-ml/tests/automl_job/unittests/test_automl_image_schema.py:127]: fixed

**Comment by novaturient95** [comment]
Can we make sure that e2e test is validated on this branch and there is no regression with API change?

**Review by MaurisLucis — APPROVED** [review]

**Review by needuv — APPROVED** [review]

**Review by novaturient95 — APPROVED** [review]

**Review by srsaggam — APPROVED** [review]

**Pull request merged**